### DISCUSSION TITLE
404 behaviour

### DISCUSSION THREAD

**Discussion by 0x241F31** [withastro/roadmap#361] [Proposal]
I want to handle SPA on path /app/*.
There is only one way to do this: define root 404.astro and inject SPA entry. But it's unobvious and also requires to load js chunk to show 404, even if current subpath is not related to app at all (imagine routes: `/app/*`, `/blog/[article]`, /blog/404).
I think Astro should search for closest 404 page in a tree if requested path wasn't found. 
Cloudflare Pages have exact behaviour: https://developers.cloudflare.com/pages/platform/serving-pages/#not-found-behavior.

That's what I've done https://github.com/MrFoxPro/astro-relative-404

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
